ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/debug/DebuggerStateManager.java

```diff
@@ -0,0 +1,150 @@
+        cleanup(workspaceServiceClient);
+        addHandlers();
+    }
+
+    /**
+     * Returns debugger state for given debugger type in scope of current workspace.
+     * 
+     * @param debuggerType
+     * @return debugger state
+     */
+    public DebugSessionStateDto getDebuggerState(String debuggerType) {
+        Map<String, DebugSessionStateDto> debugSessionStates = readStateData();
+        return debugSessionStates.get(debuggerType);
+    }
+
+    /**
+     * Sets debugger state for given debugger type in scope of current workspace.
+     * 
+     * @param debugSessionState
+     */
+    public void setDebuggerState(DebugSessionStateDto debugSessionState) {
+        Map<String, DebugSessionStateDto> debugSessionStates = readStateData();
+        debugSessionStates.put(debugSessionState.getDebuggerType(), debugSessionState);
+        writeStateData(debugSessionStates);
+    }
+
+    /**
+     * Removes debugger state for given debugger type in scope of current workspace.
+     * 
+     * @param debuggerType
+     */
+    public void removeDebuggerState(String debuggerType) {
+        Map<String, DebugSessionStateDto> debugSessionStates = readStateData();
+        debugSessionStates.remove(debuggerType);
+        writeStateData(debugSessionStates);
+    }
+
+    private void addHandlers() {
+        eventBus.addHandler(WorkspaceStoppedEvent.TYPE, new WorkspaceStoppedEvent.Handler() {
+            @Override
+            public void onWorkspaceStopped(WorkspaceStoppedEvent event) {
+                if (workspaceId.equals(event.getWorkspace().getId())) {
+                    // Wipe out storage data key on workspace stop
+                    writeStateData(null);
+                }
+            }
+        });
+    }
+
+    private Map<String, DebugSessionStateDto> readStateData() {
+        if (localStorage == null) {
+            return new HashMap<>();
+        }
+        String data = localStorage.getItem(storageUniqueKey);
+        if (data != null && !data.isEmpty()) {
+            List<DebugSessionStateDto> dssList = dtoFactory.createListDtoFromJson(data, DebugSessionStateDto.class);
+            return dssList.stream().collect(Collectors.toMap(DebugSessionStateDto::getDebuggerType, dss -> dss));
+        }
+        return new HashMap<>();
+    };
+
+    private void writeStateData(Map<String, DebugSessionStateDto> debugSessionStates) {
+        if (localStorage == null) {
+            return;
+        }
+        if (debugSessionStates == null || debugSessionStates.isEmpty()) {
+            localStorage.removeItem(storageUniqueKey);
+        } else {
+            localStorage.setItem(storageUniqueKey,
+                                 dtoFactory.toJson(debugSessionStates.values().stream().collect(Collectors.toList())));
+        }
+    }
+
+    private void cleanup(WorkspaceServiceClient workspaceServiceClient) {
+        final Storage localStorage = Storage.getLocalStorageIfSupported();

```

Comment:
Have a look on
https://github.com/eclipse/che/blob/master/ide/commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/LocalStorage.java
In recent commit I've ported some methods from `com.google.gwt.storage.client.Storage`